salt/metalk8s/volume/prepare/init.sls

```diff
@@ -0,0 +1,84 @@
+{%- set volumes = pillar.metalk8s.volumes %}
+
+{%- set volume_to_prepare = pillar['volume'] %}
+
+{%- for volume_name, volume_info in volumes.items() %}
+
+{%- if volume_to_prepare != '' %}
+# carry out the steps to provision the specified volume
+
+# check if volume exist in pillars and it is valid
+{%- if volume_to_prepare in volumes.keys() %}
+
+#check the storage class
+for v_name, v_info in volumes[volume_to_prepare]
+{%- set storageclass_name =  v_info['spec']['storageClassName']['metadata']['name'] %}
+
+#check the validity of specified volume storageclass
+#could refresh pillar here and there will be no need to check storageclass validity
+
+Verify volume storageclass:
+  metalk8s_volume.get_storageclass_validity:
+  - name: {{ storageclass_name }}
+
+    {%- if 'sparseLoopDevice' in v_info['spec'].keys() %}
+    # device to prepare is a sparsefile surely
+    # get the spec size
+{%- set v_size = volume_info['spec']['sparseLoopDevice']['size'] %}
+
+# need to convert kubernetes size object(2Gi) to human readable format
+{%- set readable_v_size = metalk8s_volume.to_bytes_converter(v_size) %}
+
+#could check here if system hard disk has sufficient disk space
+# maybe not recommended for a sparsefile
+
+Create sparseLoopDevice {{ v_name }} directory:
+  file.directory:
+  - name: /dev/{{ v_name }}
+  - user: root
+  - group: root
+  - dir_mode: 750
+  - makedirs: True
+  - require_in:
+    - file: Create sparsefile {{ v_name }}
+
+Create sparsefile {{ v_name }}:
+  cmd.run:
+  - name: "truncate --size {{ readable_v_size }} {{ v_name }}"
+  - creates: {{ v_name }}
+
+
+add sparsefile_{{ v_name }}_init_script:
+  file.managed:
+  - name: /usr/lib/systemd/system/loopdevice-{{ v_name }}.service
```

Comment:
Drop-ins don't go in `/usr`.

Reply:
Well noted

I see it should go to `/etc/systemd/system/....`